background/token-manager.js

```diff
@@ -153,17 +153,25 @@ const tokenMan = (() => {
     if (alwaysUseTab == null) {
       alwaysUseTab = await detectVivaldiWebRequestBug();
     }
+    if (hooks) hooks.query(query);
     const url = `${provider.authURL}?${new URLSearchParams(query)}`;
+    const width = Math.min(screen.availWidth - 100, 800);
+    const height = Math.min(screen.availHeight - 100, 800);
+    const wnd = await browser.windows.getLastFocused();
     const finalUrl = await webextLaunchWebAuthFlow({
       url,
       alwaysUseTab,
       interactive,
       redirect_uri: query.redirect_uri,
-      windowOptions: {
+      windowOptions: Object.assign({
         state: 'normal',
-        width: Math.min(screen.width - 100, 800),
-        height: Math.min(screen.height - 100, 800),
-      },
+        width,
+        height,
+      }, wnd.state !== 'minimized' && {
+        // Center the popup to the current window
+        top: wnd.top + (wnd.height - width) / 2,
+        left: wnd.left + (wnd.width - width) / 2,
```

Comment:
```suggestion
        top: Math.ceil(wnd.top + (wnd.height - width) / 2),
        left: Math.ceil(wnd.left + (wnd.width - width) / 2),
```

The authFlow library doesn't like decimals.